framework/src/base/MooseServer.C

```diff
@@ -0,0 +1,529 @@
+
+  app_params.set<std::vector<std::string>>("input_file") = {parse_file_path};
+
+  app_params.set<bool>("check_input") = true;
+
+  app_params.set<bool>("error_unused") = true;
+
+  app_params.set<bool>("error") = true;
+
+  app_params.set<std::string>("color") = "off";
+
+  app_params.set<bool>("disable_perf_graph_live") = true;
+
+  app_params.set<bool>("language_server") = true;
+
+  // create new application with parameters modified for input check run
+
+  this->check_app = AppFactory::instance().createShared(
+      moose_app->type(), moose_app->name(), app_params, moose_app->getCommunicator()->get());
+
+  // disable logs and enable error exceptions with initial values cached
+
+  bool cached_logging_enabled = Moose::perf_log.logging_enabled();
+
+  Moose::perf_log.disable_logging();
+
+  bool cached_throw_on_error = Moose::_throw_on_error;
+
+  Moose::_throw_on_error = true;
+
+  // run input check application converting caught errors to diagnostics
+
+  try
+  {
+    this->check_app->run();
+  }
+  catch (std::exception & err)
+  {
+    int line_number = 1;
+    int column_number = 1;
+
+    // walk over caught message line by line adding each as a diagnostic
+
+    std::istringstream caught_msg(err.what());
+
+    for (std::string error_line; std::getline(caught_msg, error_line);)
+    {
+      // skip over lines that are blank or contain only the error prefix
+
+      if (error_line.empty() || error_line == "*** ERROR ***")
+        continue;
+
+      // check if this error line already has the input file path prefix
+
+      if (error_line.rfind(parse_file_path + ":", 0) == 0)
+      {
+        // strip input file path and colon prefix off of this error line
+
+        error_line.erase(0, parse_file_path.size() + 1);
+
+        int match_line_number;
+        int match_column_number;
+        std::string match_error_line;
+
+        // get line and column number from this error line if both exist
+
+        if (pcrecpp::RE("^(\\d+)\\.(\\d+)\\-?\\d*:\\s*(.*)$")
+                .FullMatch(error_line, &match_line_number, &match_column_number, &match_error_line))
+        {
+          line_number = match_line_number;
+          column_number = match_column_number;
+          error_line = match_error_line;
+        }
+
+        // otherwise get line number off of this error line if it exists
+
+        else if (pcrecpp::RE("^(\\d+):\\s*(.*)$")
+                     .FullMatch(error_line, &match_line_number, &match_error_line))
+        {
+          line_number = match_line_number;
+          column_number = 1;
+          error_line = match_error_line;
+        }
+      }
+
+      // build diagnostic object from the error line and add to the list
+
+      diagnosticsList.push_back(wasp::DataObject());
+
+      wasp::DataObject * diagnostic = diagnosticsList.back().to_object();
+
+      // according to the protocol line and column number are zero based
+
+      pass &= wasp::lsp::buildDiagnosticObject(*diagnostic,
```

Comment:
does it make sense to continue if this step failed?

Reply:
This is a good question - that method would return false if some parameter was illegal according to the protocol.

For example, if a line number is negative, then an LSP error with an appropriate message would be communicated.

But the client, depending on implementation, could display the input diagnostics along with handling the LSP error.

So if this fails for a diagnostic, I don't think it's a bad idea to continue walking the input errors and adding the rest.